src/js/interpreter_cache.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "./interpreter_cache_interface.h"
+#include "ccf/pal/locking.h"
+#include "ds/lru.h"
+
+namespace ccf::js
+{
+  class InterpreterCache : public AbstractInterpreterCache
+  {
+  protected:
+    // Locks access to all internal fields
+    ccf::pal::Mutex lock;
+    LRU<std::string, std::shared_ptr<js::Context>> lru;
+    size_t cache_build_marker;
+
+  public:
+    InterpreterCache(size_t max_cache_size) : lru(max_cache_size) {}
+
+    std::shared_ptr<js::Context> get_interpreter(
+      js::TxAccess access,
+      const JSDynamicEndpoint& endpoint,
+      size_t freshness_marker) override
+    {
+      if (access != js::TxAccess::APP)
+      {
+        throw std::logic_error(
+          "JS interpreter reuse lru is only supported for APP "
+          "interpreters");
+      }
+
+      // Return a fresh interpreter, not stored in the cache
```

Comment:
I think this deserves a comment to clarify that we've backported this architectural change, but on `4.x` we're disabling the behaviour _exactly here_. Also worth highlighting that this is a modified merge, not a direct cherry-pick, in the initial PR description (maybe even the title).